src/react/AstarteClient.js

```diff
@@ -212,11 +212,25 @@ export default class AstarteClient {
     );
   }
 
-  registerDevice(params) {
-    const { deviceId } = params;
-    return this._post(this.apiConfig["registerDevice"](this.config), {
+  registerDevice({ deviceId, introspection }) {
+    let requestBody = {
       hw_id: deviceId
-    });
+    };
+
+    if (introspection) {
+      let encodedintrospection = {};
+
+      for (let [key, interfaceId] of introspection) {
```

Comment:
I think we could always strive to use `const` instead of `let`, to highlight that the variable won't be reassigned